lms/static/scripts/frontend_apps/utils/jwt.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * @typedef JWTPayload
+ * @prop {number} exp
+ * @prop {number} nbf
+ */
+
+/**
+ * Value class for working with JSON Web Tokens issued by the server.
+ */
+export class JWT {
+  /**
+   * Construct a JWT to wrap a recently issued JWT token.
+   *
+   * @param {string} token - Serialized JWT
+   * @param {number} issuedAt -
+   *   A _client_ timestamp in milliseconds estimating when the JWT was issued.
+   *   The estimate should bias towards being earlier than the true time, in
+   *   which case the JWT will "expire" earlier than the true expiry.
+   */
+  constructor(token, issuedAt) {
+    this._token = token;
+
+    const [, payloadBase64] = token.split('.');
+    this._payload = /** @type {JWTPayload} */ (JSON.parse(atob(payloadBase64)));
+
+    // Estimated offset of server's clock relative to client.
+    //
+    // +ve if the server's clock is ahead of the client or -ve if the server's
+    // clock is behind.
+    const skew = this._payload.nbf * 1000 - issuedAt;
```

Comment:
Maybe useful to add a comment that `nbf` stands for `not before`?
https://tools.ietf.org/html/rfc7519#page-10

Reply:
According to the standard, it maybe better if the backend generates the JWT with a `nbf` with a little margin: 

```
Implementers MAY provide for some small leeway, usually no more than a few minutes,
```

Reply:
Hmm - thinking about this it would be better to set an `iat` (issued at) timestamp on the backend rather than rely on `nbf` being the issued-at timestamp.